### PR TITLE
Support for nested links

### PR DESCRIPTION
**Pull request by bhunjadi**
Support for nested links in nested objects and arrays:

```
Coll.addLinks({
  'meta.project': {
    collection: Projects,
    type: 'one',
    field: 'meta.projectId',
  },
})
```

### CODE REVIEW

**Comment by bhunjadi** [comment]
@StorytellerCZ 

This is another PR from my repo that I didn't submit here before. I'll try to add basically everything from my fork to PRs here to have them more in sync (or ideally I'd just abandon my fork).

It's in draft because it's best to merge #403 first and then I'll make additional changes here.

**Comment by StorytellerCZ** [comment]
#403 was merged.

**Comment by bhunjadi** [comment]
@StorytellerCZ This should be ready for a review now.

File: lib/links/lib/createSearchFilters.js [hunk 1]
```diff
@@ -26,12 +26,29 @@ export default function createSearchFilters(object, linker, metaFilters) {
...
+function getIdQueryFieldStorage(object, fieldStorage, isMany = false) {
+    const [root, ...rest] = fieldStorage.split('.');
+    if (rest.length === 0) {
+        const ids = object[fieldStorage];
+        return _.isArray(ids) ? {$in: ids} : ids;
+    }
+
+    const nestedPath = rest.join('.');
+    const rootValue = object[root];
+    if (_.isArray(rootValue)) {
```
Comment (StorytellerCZ) [inline lib/links/lib/createSearchFilters.js:38]: We can use `Array.isArray`

**Review by StorytellerCZ — APPROVED** [review]
Only thing that I can pick up is an optional use of `Array.isArray` over `_.isArray`

**Comment by bhunjadi** [comment]
@StorytellerCZ Fixed in the latest commit

**Comment by StorytellerCZ** [comment]
Will be published in v1.5

**Pull request merged**